### ISSUE TITLE
process should clarify how Superseded state interacts with other maintenance process (e.g., new versions, edited/amended recommendations)

### ISSUE THREAD

**Issue by dbaron** [w3c/w3process#183]
The creation of the superseded state has caused some confusion in the CSS working group, in particular related to the discussion in w3c/csswg-drafts#2589.

My memory of the creation of superseded was that it resulted from objections to calling something obsolete that was actually superseded by a later version.

However, this has created the impression to some in the CSS WG that this creates an obligation to mark things as superseded.  The process should be clearer which sorts of revisions cause one specification to automatically be superseded by another, and which sorts require marking something as superseded.  For example, does it matter if something was an edited/amended Recommendation (versus maintenance that adds new features), or whether it had the same shortname, or whether it was produced by the same or a successor working group, or whether it has the same name (but maybe a different version)?

**Comment by dbaron** [comment]
BTW, I think one reasonable place to draw the line might be:
* if you have the ability to (reasonably) make the non-dated URL for a spec refer to the newer document, do that instead
* otherwise, the Superseded state may be appropriate

**Comment by dwsinger** [comment]
I don't think I understand what line you are trying to draw. We wanted to be able to do two things (a) clean up the /TR page so it only showed 'current' versions, not historical and (b) if someone lands on an old spec., somehow, they get alerted that it's been superseded by something newer, and that's what they should implement, consider current, etc.  Often, the new spec and the old have the same name but different version numbers, but sometimes there is a change of name (to protect the innocent) and what superceded what is not at all obvious to the non-cognoscenti of the web...

**Comment by chaals** [comment]
Noting that there is no requirement to supersede a Recommendation, I would approach it differently.

As far as I can tell, CSS 2.0 is superseded - by 2.1 - and it makes sense to republish it to make that clear.

While **bits** of CSS 2.1 are superseded by individual modules, the spec as a whole is not superseded. 

IMHO it would be helpful to publish a new Recommendation which contains the "still live" bits, points to the relevant modules, and mark the current 2.1 as superseded by that.

If 2.1 were entirely superseded by new modules, and the WG doesn't want a "core" Recommendation spec, it would also be reasonable to publish 2.1 as superseded, and note that it is superseded by some collection of specs.

In either case, this is up to the WG to decide, since they are still active.

**Comment by vfournier17** [comment]
Egads!  This seems far too complicated!  I agree with publishing a new version of a spec (2.1) and calling the prior version (2.0) superseded.

However, I wouldn't want to be the poor implementer who has to try to patch together "some collection of specs" that supersedes the prior spec (2.0).  I think the logical expectation is that each version (2.1) supersedes the version before it (2.0)

**Comment by chaals** [comment]
Lucky for you, you are not that implementor.

Reality turns out to be messy. Some implementors have actively requested that specs be broken into various pieces, others that apparently unrelated things be bundled together. Where (and if) the lines are drawn ends up being somewhat _ad hoc_... and is generally guided by input from implementors.

I think we need to recognise that, allow for the diversity of reality that Working Groups face, and trust them to make reasonable judgements.

**Comment by vfournier17** [comment]
How would you determine essential patent commitments and exclusion opportunities with a spec that's a hodgepodge of documents, or a one-off customized specification?  I understand facing reality, but patent owners probably don't want to inadvertently grant a RF patent license because a customized or bundled spec was created, without an exclusion opportunity (e.g., combination claim patents).

**Comment by dwsinger** [comment]
@vfournier17 happily we are *only* talking about slapping the label "superceded", and some text explaining where better to look, here. The actual development of the hodge-podge, mish-mash, or whatever, is part of the normal process, along with commitments, comments, and so on. 

One of the reasons for the superceded status is where Bananas 1.5 was not superceded by the obvious Bananas 2.0, but by Veg and Fruit 1.0; or vice versa, where Veg, Fruit and Cheese 10.7 was superceded by Fruits 1.0, Veg 1.0, and Dairy Products 3.0 (which in turn superceded Milk 2.0). Labels and text are needed in this case to avoid the innocent getting stuck in apparent dead-ends.

**Comment by vfournier17** [comment]
@dwsinger Got it, thanks.

**Comment by fantasai** [comment]
I would like to point out that the shortname of CSS2.0 and that of CSS2.1 are identical, i.e. 'CSS2'. (Historically, we had /TR/REC-CSS2 being 2.0 and /TR/CSS21/ being 2.1, but they were merged under the same shortname some years ago.)

**Comment by fantasai** [comment]
@chaals This means that a republication of CSS2.0 would have to be published as an update to CSS2.1... which makes no sense.

**Comment by chaals** [comment]
Hmm. https://www.w3.org/TR/REC-CSS2/ takes me to the CSS 2.0 Recommendation from 1998. https://www.w3.org/TR/CSS2/ takes me to the 2.1 Recommendation as edited in 2016. Republishing the 1998 document as superseded by the [CSS2.1 recommendation](https://www.w3.org/TR/CSS2/) doesn't seem like a problem to me - what am I missing?

**Comment by dbaron** [comment]
The CSS WG [has asked](https://log.csswg.org/irc.w3.org/css/2018-04-12/#e1004079) the team to redirect https://www.w3.org/TR/REC-CSS2/ like it also redirected https://www.w3.org/TR/CSS2/ and https://www.w3.org/TR/CSS21/ to the latest CSS 2.1.  That was just an omission.

Given that it was an omission -- is the right thing to do to ask the team to fix the "latest version" link that was missed along with all the other latest version links -- or is it to go through the superseding process?  That was part of my original question.

**Comment by dwsinger** [comment]
Yes. I think those are fine guiding practices. I think we kinda expect "latest version" to point at something later when the shortname has not changed, so formal "superseded" may not be needed (but it's harmless). The case to use the state is where otherwise the order of succession is not clear.

**Comment by dwsinger** [comment]
Is any action needed on the process document for this issue? If so, a pull request or clear edit would be nice.

**Comment by frivoal** [comment]
The Process itself doesn't speak of shortnames, and I don't think we should introduce that in there normatively just for the sake of this. However, I would suggest adding (a rephrasing of) my [earlier comment](https://github.com/w3c/w3process/issues/183#issuecomment-386865827) as a note in the process, to help people understand how this part of the Process is meant to be used.

**Comment by frivoal** [comment]
I made https://github.com/w3c/w3process/pull/213 to resolve this issue.

**Comment by frivoal** [comment]
On the process CG call, some thought PR https://github.com/w3c/w3process/pull/213 was fine, and some thought it made things even more confusing. Will put back on Agenda+ once we've reached some more agreement.

**Comment by wseltzer** [comment]
@plehegar has offered to look at the CSS questions more closely, and shared this diagram: https://w3c.github.io/tr-pages/deprecation.svg

I think we can answer without changing Process.
